controller.h

```diff
@@ -60,11 +61,19 @@ class Controller : public IController
         nlohmann::json jsonData;
         file >> jsonData;
 
-        // Deserialize the JSON into a unique_ptr<Controller>
-        unique_ptr<Controller> ptrController;
-        from_json(jsonData, ptrController);  // Explicitly use your from_json function
+        return jsonData.get<unique_ptr<Controller>>();
+    }
 
-        return ptrController;
+    void WriteToFile(const string& filePath) const override
+    {
+        // Open the file and write the JSON
+        ofstream file(filePath);
+        if (!file.is_open()) {
+            throw runtime_error("Unable to open file: " + filePath);
+        }
+
+        nlohmann::json jsonData = *this;
+        file << jsonData.dump(2);
```

Comment:
What is "2"?

Reply:
The indentation level, meaning that the file will be pretty-printed with 2 spaces per level. Which is the same level that is used in the example config.

Reply:
Maybe worth a comment as I had no idea!